src/server/game/Server/WorldSession.cpp

```diff
@@ -326,21 +327,18 @@ bool WorldSession::Update(uint32 diff, PacketFilter& updater)
             case STATUS_LOGGEDIN:
                 if (!_player)
                 {
-                    // pussywizard: such packets were sent to do something for a character that has already logged out, skip them
-                }
-                else if (!_player->IsInWorld())
-                {
-                    // pussywizard: such packets may do something important and the player is just being teleported, move to the end of the queue
-                    // pussywizard: previously such were skipped, so leave it as it is xD proper code below if we wish to change that
-
-                    // pussywizard: requeue only important packets not related to maps (PROCESS_THREADUNSAFE)
-                    /*if (opHandle.packetProcessing == PROCESS_THREADUNSAFE)
+                    // skip STATUS_LOGGEDIN opcode unexpected errors if player logout sometime ago - this can be network lag delayed packets
+                    //! If player didn't log out a while ago, it means packets are being sent while the server does not recognize
+                    //! the client to be in world yet. We will re-add the packets to the bottom of the queue and process them later.
+                    if (!m_playerRecentlyLogout)
                     {
-                        if (!firstDelayedPacket)
-                            firstDelayedPacket = packet;
+                        requeuePackets.push_back(packet);
                         deletePacket = false;
                         QueuePacket(packet);
-                    }*/
+
+                        LOG_DEBUG("network", "Re-enqueueing packet with opcode %s with with status STATUS_LOGGEDIN. "
+                                    "Player {} is currently not in world yet.", GetOpcodeNameForLogging(static_cast<OpcodeClient>(packet->GetOpcode())), GetPlayerInfo());
+                    }
                 }
                 else if (_player->IsInWorld() && AntiDOS.EvaluateOpcode(*packet, currentTime))
```

Comment:
Should we continue processing packages? How is it here TrinityCore/TrinityCore@aee4f0350ea5610b663cb07fb097f2fad256809e

Reply:
Ah yes, this is true